hyperspy/axes.py

```diff
@@ -655,6 +675,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_navigate(self):
+        self._update_attributes()
+        self._events.navigated.trigger()
+
+    def _on_nav_axes_changed(self):
+        self._update_attributes()
+        self._events.axes_changed.trigger()
+
+    def _on_axes_size_changed(self):
+        self._update_attributes()
+        self._events.axes_changed.trigger()
```

Comment:
Connecting to all the individual `DataAxis` events will potentially trigger multiple time per change, while the single `AxesManager` event will hopefully only trigger once by intelligent use of `suppress`.

Rule-of-thumb: If you are interested in the event on one specific axis, connect to that speciic axes. If you are interested in an event on all axes, use the ones on the axes manager.